lib/tasks/data_hygiene.rake

```diff
@@ -51,42 +51,65 @@ task sync_govdelivery_topic_mappings: :environment do
 
   puts "Fetching topics.."
   topics = Services.gov_delivery.fetch_topics["topics"]
+  topics = topics.map { |topic| [topic["name"], topic["code"]] }
+  subscriber_lists = SubscriberList.where.not(gov_delivery_id: nil)
+                                   .pluck(:title, :gov_delivery_id)
+                                   .map { |title, gov_delivery_id| [(title || gov_delivery_id).strip, gov_delivery_id] }
 
-  if topics.blank?
-    puts "No topics found in GovDelivery."
+  matching = topics & subscriber_lists
+  to_be_deleted = topics - subscriber_lists
+  to_be_created = subscriber_lists - topics
+
+  if to_be_deleted.blank?
+    puts "No topics to be deleted in GovDelivery."
   else
-    puts "Deleting all remote topics.."
-    topics.each do |topic|
-      puts "-- Deleting #{topic["name"]} (#{topic["code"]})"
-      Services.gov_delivery.delete_topic(topic["code"])
+    puts "Deleting remote topics.."
+    threads = to_be_deleted.each_slice(DataHygiene::BATCH_SIZE).map do |batch|
+      Thread.new do
+        batch.each do |name, code|
+          puts "-- Deleting #{name} (#{code})"
```

Comment:
Does `puts` work reliably now in threads? This suggests it used not to: http://stackoverflow.com/questions/5044299/whats-the-easiest-way-to-print-output-from-parallel-operations-in-ruby-without